### ISSUE TITLE
TypeError | Library/Configuration.php::getLdapAttributes, line 378: str_ends_with(): Argument #1 ($haystack) must be of type string, int given

### ISSUE THREAD

**Issue by cgrisu** [xperseguers/t3ext-ig_ldap_sso_auth#242]
Hey,

I just got this exception after an update from 3.7 to 4.0.2, and PHP from 7.4 to 8.2.

The method parameter is an associative array for simple field mappings: `['title' => '<dn>']`.
With my hook mapping `department = {hookName|department}` the param is a flat array: `['{hookName|department}']`

This check fails, since $field is an index:
```
foreach ($mapping as $field => $attribute) {
    if (str_ends_with($field, '.')) {
    ...
```

I checked the docs reagrding hooks, but looks like nothing has changed.

A quick fix in my case was changing this call https://github.com/xperseguers/t3ext-ig_ldap_sso_auth/blob/10429e209de3208341d9cbb599e368dca98d4153/Classes/Library/Authentication.php#L820
to `$ldapAttributes = Configuration::getLdapAttributes([$field => $value]);`

We have quite a simple setup, I didn't really test much further, but seems to work in my case.

**Comment by Nimmermaer** [comment]
Same here. 
My quick fix was cast the $field variable to a string to ensure compatibility with str_ends_with. 
This avoids potential type errors and allows the condition to work as intended.

 ` if (str_ends_with((string)$field, '.')) {
 // This is a TypoScript configuration
                    continue;
                }`

**Comment by xperseguers** [comment]
Casting to string sounds good. mind opening a PR?

**Comment by RZivaralam** [comment]
I have this problem too, and I had to use (string). Is it a bug? I used PHP 8.2 on another server with  typo3 v11, and I didn't have this problem. on the other server with typo3 v12 with the same php version, I have this error. I don't think it's because of the PHP version. in configuration.php, it used another code if (substr($field, -1) === '.') I used this one but I had the same error.

> Same here. My quick fix was cast the $field variable to a string to ensure compatibility with str_ends_with. This avoids potential type errors and allows the condition to work as intended.
> 
> ` if (str_ends_with((string)$field, '.')) { // This is a TypoScript configuration continue; }`

**Comment by xperseguers** [comment]
Yes it's a bug if you get this problem, and this may easily happen with some LDAP attribute being numeric instead of alphanumeric, so casting is safe to do, and the way to go.

**Comment by RZivaralam** [comment]
is it possible to remove  declare(strict_types=1); from the file. The old version doesn't have this code
